Add clear data notification ab test

subsymbolic opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.
-->

Task/Issue URL: https://app.asana.com/0/361428290920652/950570149935908
Tech Design URL: N/A

**Description**:
Add clear data notification experiment

**Steps to test this PR**:
**TEST PREPARATION:**
1. In VariantManager, comment out all variants other than "me", "mt", "mf", "mu"
1. Open NotificationScheduler and change `scheduleClearDataNotification(duration.toLong(), TimeUnit.DAYS)` to `scheduleClearDataNotification(30, TimeUnit.SECONDS)`
1. Open Charles and prepare to watch networking traffic

**TEST ONE: Notification launched from active app**
1. Launch the app from a clean installation
1. Ensure the notification fires after ~30 seconds
1. Check that the `notification shown` `mnot_s` pixel fires
1. Tap the notification content and ensure the settings page is opened
1. Check that the `notification launched` `mnot_l` pixel fires
1. Launch the app again and ensure the notification does not show

**TEST TWO: Notification launched from killed app**
1. Launch the app from a clean installation
1. Kill the app
1. Ensure the notification fires after ~30 seconds
1. Check that the notification shown `mnot_s` pixel fires
1. Tap the notification content and ensure the settings page is opened
1. Check that the `notification launched` `mnot_l` pixel fires
1. Launch the app again and ensure the notification does not show

**TEST THREE: Notification dismissed**
1. Launch the app from a clean installation
1. Ensure the notification fires after ~30 seconds
1. Check that the notification shown `mnot_s` pixel fires
1. Swipe the notification away to clear it
1. Check that the `notification cancelled` `mnot_c` pixel fires

**TEST FOUR: Notification not shown if clear data already set**
1. Launch the app from a clean installation
1. Update the auto clear data settings so they are not blank
1. Ensure the notification does not fire
1. Kill the app and relaunch
1. Again, ensure the notification does not fire

**TEST FIVE: Disable/Enable Notifications**
1. Disable app notifications
1. Ensure that `mnot_d` pixel fires on next launch
1. Relaunch the app and ensure the `mnot_d` pixel does not fire a second time
1. Enable app notifications
1. Ensure that `mnot_e` pixel fires on next launch
1. Relaunch the app and ensure the `mnot_e` pixel does not fire a second time

**TEST SIX: Database migration**
1. Install the version of the app in `develop` and launch it
2. Upgrade to this branch
3. Make sure nothing explodes 💥

###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

brindy commented:
1) It's not clear how to test the notification cancelling. TEST 3 seems to be the same as TEST 1?

2) I swiped the notification, tapped on settings, disabled notifications for the app and then launched the app but didn't see the cancel notification or notifications disabled pixel.  (Pixel 2, API 27)

3) I disabled notifications completely for the app (without doing it via the notification) and still don't see the notifications disabled pixel.

4) If you launch the app but don't complete the on boarding, you still get the notification. When you tap the notification and go straight to settings you can by-pass onboarding.  So when you then launch it subsequently you see the onboarding again.  Probably best not to schedule the notification if onboarding hasn't been completed?

subsymbolic commented:
Thanks for reviewing @brindy.

1. Excuse the cut and paste fail! I have updated test three to be a "swipe" away notification.
2. I see the `mnot_c` cancel pixel every time I swipe dismiss a notification. The enabled/disabled `mnot_e` & `mnot_d` weren't always showing or could be slow. I've moved them to app onResume so we seen them faster
3. As above
4. I'm ok with this for the experiment. It's an edge case and if they have installed, launched and then forgotten about us without doing the onboarding, then I want to entice them back. Will add making a smoother onboarding specific notification as a future improvement though.
